src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -76,29 +88,90 @@ private bool ResetSolutionSettingsIfNeeded()
+                // leads to multiple notifications for a single file change in the future.
+                IFileWatcher? exchanged = Interlocked.CompareExchange(ref _solutionConfigFileWatcher, newSolutionConfigFileWatcher, oldSolutionConfigFileWatcher);
+                if (ReferenceEquals(exchanged, oldSolutionConfigFileWatcher))
+                {
+                    if (newSolutionConfigFileWatcher != null)
+                    {
+                        newSolutionConfigFileWatcher.FileChanged += OnConfigFileChanged;
+                    }
+
+                    if (oldSolutionConfigFileWatcher != null)
+                    {
+                        oldSolutionConfigFileWatcher.FileChanged -= OnConfigFileChanged;
+                        oldSolutionConfigFileWatcher.Dispose();
+                    }
+                }
+                else
+                {
+                    newSolutionConfigFileWatcher?.Dispose();
+                }
+
+                ResetSolutionSettings(root);
+                return true;
+            }
+
+            return false;
+        }
+
+        // ERROR_SHARING_VIOLATION = 0x20: https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-
+        // HRESULT_FROM_WIN32 = 0x80070000 | (code & 0xffff): https://learn.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-hresult_from_win32
+        private const uint WindowsSharingViolationHResult = 0x80070020;
+
+        private void ResetSolutionSettings(string? solutionDirectory)
+        {
+            _solutionSettings = new Tuple<string?, AsyncLazy<ISettings>>(
+                item1: solutionDirectory,
+                item2: new AsyncLazy<ISettings>(async () =>
+                {
+                    ISettings settings;
+                    try
+                    {
+                        ISettings? loadedSettings = null;
+                        int retryCount = 0;
+                        while (loadedSettings == null)
                         {
-                            ISettings settings = null;
                             try
                             {
-                                settings = Settings.LoadDefaultSettings(root, configFileName: null, machineWideSettings: MachineWideSettings);
+                                loadedSettings = Settings.LoadDefaultSettings(solutionDirectory, configFileName: null, machineWideSettings: MachineWideSettings);
                             }
+                            // If any config files are in use, retry after a short delay
+                            // Would be nice if there was a better way to detect file busy: https://github.com/dotnet/runtime/issues/79643
                             catch (NuGetConfigurationException ex)
+                                when (ex.InnerException?.GetType() == typeof(IOException)
+                                && (uint)ex.InnerException.HResult == WindowsSharingViolationHResult
```

Comment:
Why do you need to cast to unit? Can you save the cast by comparing against an `int` and defining `WindowsSharingViolationHResult` as a `const int` ?

```suggestion
                                && ex.InnerException.HResult == WindowsSharingViolationHResult
```

Reply:
Because casting 0xF0070020 to `int` doesn't work (this surprised me, I thought it was possible), and it was easier than calculating what the int value is in signed base-10. Plus, keeping the hex value makes it easier to figure out what the value is, why it has the value it does. Converting to signed base 10 just to remove a cast in my opinion obscures the value and therefore reduces readability.